CONTRIBUTING.md

```diff
@@ -0,0 +1,99 @@
+This page explains common tasks related to working with Xtext's source code.
+
+## Report an Issue
+
+Please create a GitHub issue in this repository.
+
+Eclipse Bugzilla issues are deprecated for Xtext. However, we still accept them.
+
+### Milestones
+
+We have one milestone for each Xtext release. Fixed issues should be assigned to the milestone of the nearest release in which the fix will be included. For example, if an issue is fixed on the `maintenance` branch, the corresponding milestone is the next service release (increasing the third version component). If an issue is fixed on the `main` branch, the next major or minor release should be assigned.
+
+It is possible to assign issues to a milestone before they are fixed, which can be useful for planning. However, make sure someone will actually work on that issue when you do that!
+
+The list of issues assigned to a milestone will give a nice overview of new features and fixed bugs after a release has been done, facilitating the creation of release notes and helping users to know when a fix will be available.
+
+### Issue Labels
+
+Issue labels can serve several purposes:
+
+* Indicate the issue type: `bug`, `enhancement`, `new_feature`, `question`
+* Communicate the current status of an issue: `confirmed`, `help_wanted`
+* Communicate why an issue has not been fixed: `duplicate`, `invalid`, `wontfix`
+* Categorize issues for better overview, e.g. to assign an issue to a specific part of the software. Committers may add categorization labels as required. However, in most cases the `[category]` prefixing in the issue title should be sufficient.
+
+In general we want to keep the number of labels as low as possible in order to avoid contributors to be overwhelmed and to make sure they are actually used. A label is not useful if it's assigned only to a fraction of the issues it belongs to, so it's important for all contributors to use the existing labels consistently.
```

Comment:
```suggestion
In general, we want to keep the number of labels as low as possible to avoid overwhelming contributors and ensure the labels are used. A label is not helpful if it is assigned only to a fraction of the issues it belongs to, and that's why all contributors need to use the existing labels consistently.
```